src/lib.rs

```diff
@@ -197,8 +203,34 @@ fn setup_sqlite(
         None
     };
 
+    // If opening an existing wallet whose database does not yet store the wallet transactions,
+    // query them from the Bitcoin backend before proceeding to the migration.
     let sqlite = SqliteDb::new(db_path, options, secp)?;
-    sqlite.maybe_apply_migrations()?;
+    if !fresh_data_dir {
+        let mut conn = sqlite.connection()?;
+        let wallet_txs = if conn.db_version() <= MAX_DB_VERSION_NO_TX_DB {
+            let bit = bitcoind.as_ref().ok_or(StartupError::DbMigrateBitcoinTxs(
+                "a connection to a Bitcoin backend is required",
```

Comment:
nit: "to bitcoind". This code is bitcoind-specific (on purpose).

Reply:
I've left these as they are for now but I see your point.